src/commands/external.rs

```diff
@@ -123,6 +88,98 @@ pub async fn execute_external_subcommand(
     Ok(())
 }
 
+async fn ensure_plugin_available(
+    plugin_name: &str,
+    plugin_store: &PluginStore,
+    app: clap::App<'_>,
+    override_compatibility_check: bool,
+) -> anyhow::Result<Option<String>> {
+    let plugin_version = match plugin_store.read_plugin_manifest(plugin_name) {
+        Ok(manifest) => {
+            if let Err(e) =
+                warn_unsupported_version(&manifest, SPIN_VERSION, override_compatibility_check)
+            {
+                eprintln!("{e}");
+                // TODO: consider running the update checked?
```

Comment:
Hmmm, that could be handy.  But potentially also another spot to check in with user 'no plugin manifest is found for foo, shall we update your local plugins list'?  Though I think just proceeded seems like a sane default eg 'no plugin manifest is found for foo, pulling the latest available plugins from fermyon/spin-plugins' (or other).